llap-tez/src/java/org/apache/hadoop/hive/llap/tezplugins/LlapTaskCommunicator.java

```diff
@@ -190,31 +173,46 @@ public LlapTaskCommunicator(
     }
   }
 
+  @SuppressWarnings("unchecked")
+  private Token<LlapTokenIdentifier> getLlapToken() {
+    Token<LlapTokenIdentifier> token = null;
+    Credentials credentials = getContext().getAMCredentials();
+    if (credentials != null) {
+      token = (Token<LlapTokenIdentifier>) credentials.getToken(LlapTokenIdentifier.KIND_NAME);
+    }
+    Preconditions.checkState((token != null) == UserGroupInformation.isSecurityEnabled());
+    LOG.info("Task communicator with a token {}", token);
```

Comment:
The Precondition is pretty deep, took me a second to understand, can of comment telling above, that there should be a token if security is enabled
BTW. if security is disabled the log line will log the token and as as ``null``

Reply:
agree, putting a comment:
```
// supposed to have a token only if security is enabled (any mismatch here implies a configuration problem)
```